src/function/table/system/duckdb_functions.cpp

```diff
@@ -212,12 +216,70 @@ struct MacroExtractor {
 		return Value::LIST(LogicalType::VARCHAR, move(results));
 	}
 
-	static Value GetVarArgs(MacroCatalogEntry &entry, idx_t offset) {
+	static Value GetVarArgs(ScalarMacroCatalogEntry &entry, idx_t offset) {
 		return Value();
 	}
 
-	static Value GetMacroDefinition(MacroCatalogEntry &entry, idx_t offset) {
-		return entry.function->expression->ToString();
+	static Value GetMacroDefinition(ScalarMacroCatalogEntry &entry, idx_t offset) {
+		if (entry.function->type == MacroType::SCALAR_MACRO) {
+			auto &func = (ScalarMacroFunction &)*entry.function;
+			return func.expression->ToString();
+		}
+		return Value();
+	}
+};
+
+struct TableMacroExtractor {
+	static idx_t FunctionCount(TableMacroCatalogEntry &entry) {
```

Comment:
This never appears to be used either, could you add a test where you run `duckdb_functions` after creating several table macro function in the catalog?